### PR TITLE
Add optional annotations parameter to `import-dictionary`

### PR DESCRIPTION
**Pull request by EduardSergeev**
Implements "import-dictionary" part of https://github.com/ambiata/ivory/issues/720

### CODE REVIEW

File: ivory-operation/src/main/scala/com/ambiata/ivory/operation/ingestion/DictionaryImporter.scala [hunk 1]
```diff
@@ -47,5 +50,5 @@ object DictionaryImporter {
   case object Update extends ImportType
   case object Override extends ImportType
 
-  case class ImportOpts(ty: ImportType, force: Boolean)
+  case class ImportOpts(ty: ImportType, force: Boolean, annotation: Option[Annotation])
```
Comment (charleso) [inline ivory-operation/src/main/scala/com/ambiata/ivory/operation/ingestion/DictionaryImporter.scala]: Would we want/need multiple annotations?

Comment (charleso) [inline ivory-operation/src/main/scala/com/ambiata/ivory/operation/ingestion/DictionaryImporter.scala]: Just ignore me...

**Comment by charleso** [comment]
Looks good so far :+1:

File: ivory-operation/src/main/scala/com/ambiata/ivory/operation/ingestion/DictionaryImporter.scala [hunk 2]
```diff
@@ -36,8 +38,9 @@ object DictionaryImporter {
       doImport = validation.isSuccess || importOpts.force
       dictIdIden <- if (doImport)
         for {
+          annIdO <- importOpts.annotation.traverseU(a => RepositoryT.runWithRepo(repository, AnnotationTextStorage.store(a)))
```
Comment (charleso) [inline ivory-operation/src/main/scala/com/ambiata/ivory/operation/ingestion/DictionaryImporter.scala]: Should this be merging annotations?

Comment (EduardSergeev) [inline ivory-operation/src/main/scala/com/ambiata/ivory/operation/ingestion/DictionaryImporter.scala]: Yes, if `--merge` flag is supplied. Still WIP, will add it before PR is fully ready for review.

**Comment by EduardSergeev** [comment]
I am leaving here only "import-dictionary" part of the task.

Other commands will be extended in separate PR's if needed.

**Comment by EduardSergeev** [comment]
Bump @charleso I believe this is it for `import-dictionary`
OK to merge?

File: ivory-cli/src/main/scala/com/ambiata/ivory/cli/annotation/Args.scala [hunk 3]
```diff
@@ -2,15 +2,21 @@ package com.ambiata.ivory.cli.annotation
...
-import pirate._, Pirate._
+import com.ambiata.ivory.operation.annotation.AnnotationManager._
+import pirate._, pirate.Pirate._
 import scalaz._, Scalaz._
 
 object Args {
 
+  def annotationArgs: Parse[AnnotationOpts] =
+      ( annotationDef
+    |@| switch(long("merge"), description("Merge given annotation with existing one instead of replacing it"))
```
Comment (charleso) [inline ivory-cli/src/main/scala/com/ambiata/ivory/cli/annotation/Args.scala]: I'm wondering whether merge should be the default or not?

**Comment by charleso** [comment]
:+1:

File: ivory-cli/src/main/scala/com/ambiata/ivory/cli/annotation/Args.scala (hunk 3 above)
Comment (EduardSergeev) [inline ivory-cli/src/main/scala/com/ambiata/ivory/cli/annotation/Args.scala]: Now I am wondering too :) Previously I though that `replace` would be more sensible behaviour but now that you've mentioned it I am no longer sure. Do we have any other potential usages on annotations in mind? If there is only "napoleon-commit" key it doesn't matter but with other annotations in place `merge` may be more convenient default. @markhibberd to the rescue?

Comment (markhibberd) [inline ivory-cli/src/main/scala/com/ambiata/ivory/cli/annotation/Args.scala]: I think the fact that replace implies you know what all the other annotations are, means it isn't the best default. I have some ideas for other annotations, but I don't think that it matters, the chance that two independent jobs might accidentally clobber each others annotations if they don't know about each other, makes merge seem like the only reasonable default. (Replace may only be useful if you were doing housekeeping / maintenance and wanted to clear out old annotations or similar maybe)

Comment (EduardSergeev) [inline ivory-cli/src/main/scala/com/ambiata/ivory/cli/annotation/Args.scala]: Yes, makes total sense. Merging by default and `--replace` flag is added.

**Pull request merged**